src/praisonai-agents/praisonaiagents/goal/loop.py

```
@@ -0,0 +1,211 @@
+            return
+        store = getattr(self, "_session_store", None) or getattr(
+            self, "session_store", None
+        )
+        if store is None:
+            return
+        try:
+            store.update_session_metadata(
+                self._goal_state_key().split(":", 1)[1],
+                goal_state=state.to_dict(),
+            )
+        except Exception as exc:  # pragma: no cover - persistence is best-effort
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)